ecl/hql/hqlfold.cpp

```diff
@@ -2154,6 +2154,74 @@ IHqlExpression * foldConstantOperator(IHqlExpression * expr, unsigned foldOption
             }
             break;
         }
+    case no_regex_findset:
+        {
+            IValue * t0 = expr->queryChild(0)->queryValue();
+            IValue * t1 = expr->queryChild(1)->queryValue();
+            if (t0 && t1)
+            {
+                if(isUnicodeType(t0->queryType()))
+                {
+                    unsigned plen = t0->queryType()->getStringLen();
+                    OwnedPtr<UChar> pattern (new UChar[plen+1]);
+                    t0->getUCharStringValue(plen+1, pattern.get()); //plen+1 so get null-terminated
+
+                    unsigned slen = t1->queryType()->getStringLen();
+                    OwnedPtr<UChar> search (new UChar[slen]);
+                    t1->getUCharStringValue(slen, search);
+
+                    ICompiledUStrRegExpr * compiled = rtlCreateCompiledUStrRegExpr(pattern.get(), !expr->hasAttribute(noCaseAtom));
+                    assertex(compiled!=NULL);
+
+                    HqlExprArray results;
+                    IUStrRegExprFindInstance * match = compiled->find(search, 0, slen);
+                    while (match->found())
```

Comment:
It is changing the semantics.  Previously once the object had been created the calls to getMatchX() would always return the same thing.  With this change it is no longer the case because nextMatch() modifies the state of the object.